src/LiveQuery/ParseLiveQueryServer.js

```diff
@@ -581,6 +581,21 @@ class ParseLiveQueryServer {
       });
   }
 
+  async getAuthFromClient(client: any, requestId: number, sessionToken: string) {
+    const getSessionFromClient = () => {
+      const subscriptionInfo = client.getSubscriptionInfo(requestId);
+      if (typeof subscriptionInfo === 'undefined') {
+        return client.sessionToken;
+      }
+      return subscriptionInfo.sessionToken || client.sessionToken;
+    };
+    if (!sessionToken) {
+      sessionToken = getSessionFromClient();
+    }
+    const { auth } = await this.getAuthForSessionToken(sessionToken);
+    return auth || {};
```

Comment:
In which case is `auth === undefined`? Following up, that would also set `request.user = undefined`, so I wonder what that means.

Reply:
Would you prefer:

```
return auth || {};
if (auth && auth.user) {
  res.user = auth.user;
}
```

> In which case is auth === undefined?

Not sure, but without it, `cannot read property user of undefined` will occasionally throw.

Reply:
> Not sure, but without it, cannot read property user of undefined will occasionally throw.

I think we'd have to understand why this can be `undefined` first to understand how to handle this.

Reply:
I’m not entirely sure how the internal `Auth` works. I’m assuming that if an invalid session is passed through, it will return undefined.

Reply:
I think you could find out by going backwards until you find maybe a query or some validation that returns `undefined`. Maybe there are also existing tests that give a hint.

Reply:
All the tests for `getAuthForSessionToken` are testing for `userAuth.user`.

Why it returns `undefined` is because the LQ's inbuilt `getAuth` method, that then calls `auth`'s `getAuth` method:

```
getAuthForSessionToken(sessionToken: ?string): Promise<{ auth: ?Auth, userId: ?string }> {
    if (!sessionToken) {
      return Promise.resolve({});
    }
    ....
}
```

If a session token isn't provided, it returns `{}`. Which means that the destructured `const { auth } = await this.getAuthForSessionToken(sessionToken);` is undefined (it's looking for {}.auth).

In that case, would it be fine to do:

`return auth || {}`
```
if (auth.user) {
    res.user = auth.user;
}
```

Couple of notes:
- If an invalid session is passed, `Invalid session token` is thrown. This is consistent with other uses of `getAuthForSessionToken` in this file.
- The "auth" object has to be returned so that `roleNames` can be inferred from auth for the cloud validator `requireUser roles`

Reply:
> If a session token isn't provided, it returns {}. 

> If an invalid session is passed, Invalid session token is thrown. 

Should we throw also on missing token instead of passing undef?

Reply:
I don’t think so, as missing token is the use case for no user logged in